common/etl/buffers.go

```diff
@@ -0,0 +1,167 @@
+package etl
+
+import (
+	"bytes"
+	"sort"
+	"strconv"
+)
+
+const (
+	//SliceBuffer - just simple slice w
+	SortableSliceBuffer = iota
+	//SortableAppendBuffer - just simple slice w
+	SortableAppendBuffer
+
+	BufferOptimalSize = 256 * 1024 * 1024 /* 256 mb | var because we want to sometimes change it from tests */
+)
+
+type Buffer interface {
+	Put(k, v []byte)
+	Get(i int) sortableBufferEntry
+	Len() int
+	Reset()
+	GetEntries() []sortableBufferEntry
+	Sort()
+	CheckFlushSize() bool
+}
+
+type sortableBufferEntry struct {
+	key   []byte
+	value []byte
+}
+
+var (
+	_ = &sortableBuffer{}
+	_ = &appendSortableBuffer{}
+)
+
+func NewSortableBuffer(bufferOptimalSize int) *sortableBuffer {
+	return &sortableBuffer{
+		entries:     make([]sortableBufferEntry, 0),
+		size:        0,
+		optimalSize: bufferOptimalSize,
+	}
+}
+
+type sortableBuffer struct {
```

Comment:
I really like the idea that our "collection" behaviour differs dependent on a buffer type. We might need one more later on.
Though, I'm not sure if this old `sortableBuffer` should be used for any reason to be honest.
I think we can safely replace it with a map that you did as `appendBuffer`

Reply:
It's ~5%faster on flat data(on my laptop). I decided to keep it.